app/models/company.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+# Model that allows storing companies
+class Company < ApplicationRecord
+  before_save :ensure_only_one_selected
+
+  validates :name, :website, :selected, presence: true
+
+  has_many :social
```

Comment:
```suggestion
  has_many :socials
```

If it's many, it makes sense to be plural, no?